lib/git.js

```diff
@@ -218,12 +218,36 @@ export async function verifyAuth(repositoryUrl, branch, execaOptions) {
  *
  * @param {String} tagName The name of the tag.
  * @param {String} ref The Git reference to tag.
+ * @param {Object} tagOptions Tag config options.
  * @param {Object} [execaOpts] Options to pass to `execa`.
  *
  * @throws {Error} if the tag creation failed.
  */
-export async function tag(tagName, ref, execaOptions) {
-  await execa("git", ["tag", tagName, ref], execaOptions);
+async function tag(tagName, ref, {tagAnnotate, tagSign, tagMessage}, execaOptions) {
+  const args = ["tag", tagName, ref];
+  const gitTagSign = await shouldSignTag(execaOptions);
```

Comment:
What is the reasoning behind accounting for & giving precedence to the git configuration?

I see a few problems with that:
- if I set `tagSign` to `false` in my config, it might still sign the tag based on my system configuration
- it makes the changes breaking, as it will start unsolicitedly signing tags

Because of that, I don't think the git configuration should be accounted for. If overriding the release configuration behavior is needed in a specific setup, overriding it with the CLI flag is enough.

Reply:
If I remember correctly...
I check for the git config first because signed tags **must** have a message, otherwise it fails. Currently if I set `tag.gpgsign` to `true` it will fail when creating a tag.
So if you set `tagSign` to `false` and `tag.gpgsign` is `true` it will still sign a tag to prevent the release from failing.
That said I don't think it is unsolicitedly if I previously set `tag.gpgsign`.
Does this make sense?